### PR TITLE
Dataproc submit job operator async

### PR DESCRIPTION
**Pull request by bjankie1**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/google/cloud/hooks/dataproc.py [hunk 1]
```diff
@@ -227,7 +237,10 @@ def get_job_client(self, region: Optional[str] = None) -> JobControllerClient:
             client_options = ClientOptions(api_endpoint=f'{region}-dataproc.googleapis.com:443')
 
         return JobControllerClient(
-            credentials=self._get_credentials(), client_info=CLIENT_INFO, client_options=client_options
+            credentials=self._get_credentials(),
+            client_info=CLIENT_INFO,
+            client_options=client_options,
+            transport="grpc_asyncio" if self.async_client else "grpc",
```
Comment (lwyszomi) [inline airflow/providers/google/cloud/hooks/dataproc.py:243]: In the Dataproc SDK we have `JobControllerAsyncClient` so maybe we should consider to use it: https://github.com/googleapis/python-dataproc/blob/main/google/cloud/dataproc_v1/services/job_controller/async_client.py#L43

File: airflow/providers/google/cloud/triggers/dataproc.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                "region": self.region,
+                "gcp_conn_id": self.gcp_conn_id,
+                "delegate_to": self.delegate_to,
+                "impersonation_chain": self.impersonation_chain,
+                "pooling_period_seconds": self.pooling_period_seconds,
+            },
+        )
+
+    async def get_job(self) -> Job:
+        return await self.hook.get_job(job_id=self.job_id, project_id=self.project_id, region=self.region)
```
Comment (lwyszomi) [inline airflow/providers/google/cloud/triggers/dataproc.py:56]: this code is not used so we can remove it

File: tests/providers/google/cloud/operators/test_dataproc.py [hunk 3]
```diff
@@ -61,6 +63,7 @@
 AIRFLOW_VERSION = "v" + airflow_version.replace(".", "-").replace("+", "-")
 
 DATAPROC_PATH = "airflow.providers.google.cloud.operators.dataproc.{}"
+COMPOSER_TRIGGERS_STRING = "airflow.providers.google.cloud.triggers.dataproc.{}"
```
Comment (lwyszomi) [inline tests/providers/google/cloud/operators/test_dataproc.py:66]: should be "DATAPROC"?

File: airflow/providers/google/cloud/hooks/dataproc.py (hunk 1 above)
Comment (bjankie1) [reply airflow/providers/google/cloud/hooks/dataproc.py]: Thank you for the suggestion. Done

File: airflow/providers/google/cloud/triggers/dataproc.py (hunk 2 above)
Comment (bjankie1) [reply airflow/providers/google/cloud/triggers/dataproc.py]: Good catch!

File: tests/providers/google/cloud/operators/test_dataproc.py (hunk 3 above)
Comment (bjankie1) [reply tests/providers/google/cloud/operators/test_dataproc.py]: Correct! Fixed

**Review by lwyszomi — APPROVED** [review]

**Pull request closed without merging**